src/name_service/storage/mongo.clj

```diff
@@ -1,4 +1,46 @@
-(ns name-service.storage.mongo)
+(ns name-service.storage.mongo
+  (:require [clojure.spec.alpha :as s]
+            [clojure.walk :as w]
+            [environ.core :refer [env]]
+            [monger.collection :as mc]
+            [monger.core :as mg]
+            [name-service.core.entity :as entity]
+            [name-service.core.spec :as cs]))
 
-(defn make-storage []
-  {})
\ No newline at end of file
+(defn- fetch
+  [{:keys [conn coll-name]} key val]
+  (-> (mc/find-one-as-map conn coll-name {key val})
+      (dissoc :_id)
+      w/stringify-keys))
+
+(defn- save
+  [{:keys [conn coll-name] :as storage} keymap key val]
+  (if (s/valid? ::cs/keymap keymap)
+    (-> (if (empty? (fetch storage key val))
+          (mc/insert-and-return conn coll-name keymap)
+          (mc/find-and-modify conn
+                              coll-name
+                              {key val}
+                              keymap
+                              {:return-new true}))
+        (dissoc  :_id)
+        w/stringify-keys)
+    :not-valid))
+
+(defn- delete
+  [{:keys [conn coll-name]} key val]
+  (if (and (s/valid? ::cs/key key) (s/valid? ::cs/val val))
+    (mc/remove conn coll-name {key val})
+    :not-valid))
+
+(defrecord MongoStorage [db]
+  entity/KeyBindingStorage
+  (-fetch [_ key val] (fetch db key val))
+  (-save [_ keymap key val] (save db keymap key val))
+  (-delete [_ key val] (delete db key val)))
+
+(defn make-storage [coll-name uri]
+  (let [db {:coll-name coll-name}
+        conn (:db (mg/connect-via-uri uri))]
+    (->MongoStorage {:conn conn
+                     :coll-name (:coll-name db)})))
```

Comment:
I think this let it totally unneded here. You could do something like this instead:
```suggestion
(defn make-storage [coll-name uri]
    (->MongoStorage {:conn (-> (mg/connect-via-uri uri) :db)
                     :coll-name coll-name}))
```

Also I would rather pass `:db` keyword instead of `:conn`